app/decorators/manageiq/providers/hawkular/middleware_manager_decorator.rb

```diff
@@ -7,5 +7,24 @@ def self.fonticon
     def self.fileicon
       "svg/vendor-hawkular.svg"
     end
+
+    def quadicon(_n = nil)
+      {
+        :top_left     => {
+          :text => middleware_domains.size
+        },
+        :top_right    => {
+          :text => middleware_servers.size
+        },
+        :bottom_left  => {
+          :fileicon => fileicon,
+          :tooltip  => type
+        },
+        :bottom_right => {
```

Comment:
can the bottom right part = auth status be extracted as a default? it seems repeating

Reply:
@abonas I agree with you as I'm a fan of the DRY philosophy, but decorators are an exception. Think about them as a place to specify the UI sugar for each DB model. Eventually it will get simplified to a DSL that can be edited by designers:
```
class SomethingDecorator
  quadicon :top_left, :fileicon => '100/something.png', :tooltip => 'yay'
  quadicon :top_right, :text => 3
end
```
I think it's understandable that we don't want to pull shared definitions on such places...